### ISSUE TITLE
A11y: The focus order does not match the visual position

### ISSUE THREAD

**Issue by neet** [xyflow/xyflow#5189]
### Please describe the feature that you want to propose

Hi, thank you for maintaining an awesome library. I really love its intuitive APIs and interfaces.

Recently, our team conducted an accessibility test against our company's products, and we discovered some obstacles for screen readers. We noticed that some of the problems we discovered originated from Rect Flow.

One of the most critical issues we observed is that the order of elements we can focus on via both <kbd>Tab</kbd> or <kbd>Arrow</kbd>s does not match the visual position of the elements:

https://github.com/user-attachments/assets/6b7f3843-e593-414b-9afc-07a27e958e62

As you can see on the clip, it focuses edges first and the nodes last. It would be great if we could focus on nodes and their belonging edges first, and then move the sibling nodes, which is the same as the experience it offers to visual users.

I believe this violates [2.4.3 Focus Order from WCAG 2.1](https://www.w3.org/TR/WCAG21/#focus-order). It is one of the Level A criteria, the ones that have the most significant impact on users who need assistive technology.

I recorded the video with VoiceOver in macOS 15.3.2 and the latest Chrome.

**Comment by msftedad** [comment]
Hi @moklick, Any update on this issue?

**Comment by moklick** [comment]
For now we decided against adding this to the core library. We might add an example for this at some point.

**Comment by kesteer** [comment]
@moklick Just chiming in here as we have a similar issue in our application. Do you mind sharing the reasoning behind the decision against adding this to the core library? Is it a matter of resources or rather a more fundamental design decision thats causing an issue with implementation.

**Comment by jendowns** [comment]
Hello! Adding my voice as well: my team would also love to see this issue resolved in the core library. Are you open to external contributions to fix this? 

Thank you for your time.

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?
